Css/WebScrap.Css/CssComplianceChecker.cs

```diff
@@ -1,45 +1,110 @@
 using WebScrap.Common.Tags;
 using WebScrap.Css.Listeners.Helpers;
+using WebScrap.Css.Preprocessing.Tokens;
 
 namespace WebScrap.Css;
 
-internal class CssCompliantChecker(
-    Stack<OpeningTag> traversedTags, 
-    Stack<OpeningTag> cssCompliantTags)
+// TODO: its unclear how its working.
+// And its a legit drunk code.
+internal class CssComplianceChecker
 {
-    internal bool CheckLength()
-        => cssCompliantTags.Count <= traversedTags.Count;
+    private readonly Stack<CssTokenBase> cssCompliantTags;
+    private readonly Stack<OpeningTag> traversedTags;
 
-    internal bool CheckNames()
+    public CssComplianceChecker(
+        Stack<CssTokenBase> cssCompliantTags,
+        Stack<OpeningTag> traversedTags)
     {
-        var cssTags = new Stack<OpeningTag>(cssCompliantTags.Reverse());
-        var travTags = new Stack<OpeningTag>(traversedTags.Reverse());
-        while (cssTags.Count != 0)
+        this.cssCompliantTags = cssCompliantTags;
+        this.traversedTags = traversedTags;
+    }
+
+    internal static bool CheckLength(CssComplianceChecker checker)
+    {
+        var clone = Clone(checker);
+        return clone.cssCompliantTags.Count <= clone.traversedTags.Count;
+    }
+
+    internal static bool CheckNames(CssComplianceChecker checker) 
+    {
+        var clone = Clone(checker);
+        return clone.Traverse((cssTag, travTag)
+            => cssTag.Name.SequenceEqual(travTag.Name));
+    }
+
+    internal static bool CheckAttributes(CssComplianceChecker checker)
```

Comment:
> CheckXXX method family looks suitable here, but

I see it better if the Traverse -> isValid call will be reversed.

The `css.Name.Equals(Name)` actually belongs here, 
but the wrapper `clone.Traverse` is alien here.

Reply:
![image](https://github.com/Alex-Kozachenko/WebScrap/assets/119358312/cfdd3b93-7a35-4c2c-b8a5-35f55b227b1f)